src/ws/service/friendship_event_handler.rs

```diff
@@ -0,0 +1,139 @@
+use std::sync::Arc;
+
+use crate::{
+    components::database::DatabaseComponentImplementation,
+    ws::{
+        app::SocialContext,
+        service::{
+            database_handler::{get_friendship, get_last_history, update_friendship_status},
+            errors::{FriendshipsServiceError, FriendshipsServiceErrorResponse},
+            types::{EventResponse, FriendshipPortsWs, RoomInfoWs},
+        },
+    },
+    UpdateFriendshipPayload,
+};
+
+use super::{
+    friendship_event_validator::validate_new_event,
+    friendship_status_calculator::get_new_friendship_status,
+    mapper::extract_update_friendship_payload,
+    synapse_handler::{
+        get_or_create_synapse_room_id, set_account_data, store_message_in_synapse_room,
+        store_room_event_in_synapse_room,
+    },
+};
+
+pub async fn process_room_event(
```

Comment:
can we add a comment here explaining what processing a room means? (and/or change the method name to a more explicit one)